### PR TITLE
Initialize keyvaluestore

### PR DESCRIPTION
**Pull request by tkoivi**
Initialize kvs based on blog post

Tested with curl commands demonstrated in post.

### CODE REVIEW

**Comment by louislef299** [comment]
Just need to rebase with main to fix that merge conflict. You can do that by running:

```bash
git checkout main
git pull
git checkout feature/init-kvs
git rebase main
```

**Review by defoeam — CHANGES_REQUESTED** [review]
One quick change: can you change the "keyvaluestore" package name to "kv"?
It will be "github.com/defoeam/kvs/kv" in main.go.

**Comment by louislef299** [comment]
Could we also write the [tests in Go](https://go.dev/wiki/TableDrivenTests)?

**Comment by defoeam** [comment]
> Could we also write the [tests in Go](https://go.dev/wiki/TableDrivenTests)?

Yeah for sure! I haven't tried table-driven tests yet but this looks promising

**Review by defoeam — DISMISSED** [review]

File: main.go [hunk 1]
```diff
@@ -1,7 +1,89 @@
...
+	writeTimeout := time.Duration(timeoutMultiplier) * time.Second
+	server := &http.Server{
+		Addr:         addr,
+		ReadTimeout:  readTimeout,
+		WriteTimeout: writeTimeout,
+	}
+
+	log.Printf("Starting key-value store on http://localhost%s\n", addr)
+
+	err := server.ListenAndServe()
```
Comment (louislef299) [inline main.go:85]:
```suggestion
	log.Fatal(server.ListenAndServe())
```

nit: You can just wrap the listen and serve function in a log.Fatal error. Feel free to close this if you would like to just get this merged.

Comment (defoeam) [reply main.go]: Fixed

**Review by defoeam — DISMISSED** [review]

**Review by louislef299 — APPROVED** [review]

**Pull request merged**